src/icalendar/parser.py

```diff
@@ -181,28 +185,47 @@ def q_join(lst, sep=",", always_quote=False):
     return sep.join(dquote(itm, always_quote=always_quote) for itm in lst)
 
 
-def single_string_parameter(func):
-    """Create a parameter getter/setter for a single string parameter."""
+def single_string_parameter(func: Callable | None = None, upper=False):
```

Comment:
This function doesn't appear in the docs where I expect it.

https://icalendar--979.org.readthedocs.build/en/979/reference/api/icalendar.parser.html

Reply:
That should not be in the docs. It is not exported. It is private.

Reply:
The issue can take this function as an example and have another check that all functions that are not exported with from icalendar import xxx should be make private.